client/cmd/dexc-desktop/pkg/pkg-darwin.sh

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
+
+# Bail out on any unhandled errors
+set -e;
+
+# Any command that exits with non-zero code will cause the pipeline to fail
+set -o pipefail;
+
+export GOWORK=off
+
+# For release set metadata to "release".
+VER=`cat ../../../app/VERSION`
+META= # "release"
+BUILD_VER="1.0.0" # increment for every build.
+OS_FULL_VERSION="$(sw_vers | sed -n 2p | cut -d : -f 2 | tr -d '[:space:]' | cut -c1-)"
+OS_MAJOR_VERSION="$(echo $OS_FULL_VERSION | cut -d . -f 1)"
+OS_MINOR_VERSION="$(echo $OS_FULL_VERSION | cut -d . -f 2)"
+
+# if META set, append "+${META}", otherwise nothing.
+LDFLAGS_BASE="-buildid= -s -w -X main.Version=${VER}${META:++${META}}"
+
+# App information
+APP_NAME="Decred DEX"
+VOLUME_NAME="Decred DEX ${VER}${META:++${META}}"
+
+# Filepaths to important directories.
+SRC_DIR="$(cd ../src && pwd && cd ../pkg)"
+SCRIPTPATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+INSTALLERS_DIR="$SCRIPTPATH/installers"
+APP_DIR="${SCRIPTPATH}/${APP_NAME}.app"
+CONTENTS_DIR="${APP_DIR}/Contents"
+RESOURCES_DIR="${CONTENTS_DIR}/Resources"
+APP_EXCE_DIR="${CONTENTS_DIR}/MacOS"
+
+# For fancy disk image container icon and background.
+ICON_FILE_NAME=dexc-icon.icns
+ICON_FILE="${SRC_DIR}/dexc-icon.icns"
```

Comment:
Our icon doesn't seem to be standardized.
![image](https://github.com/decred/dcrdex/assets/6109680/d2033094-b347-4883-b868-1ebebd0b0c2d)

Reply:
Kinda.

> We might have to create icons of various sizes:
16x16 pixels (1x and 2x), 32x32 pixels (1x and 2x), 64x64 pixels (1x and 2x), 128x128 pixels (1x and 2x), 256x256 pixels (1x and 2x), 512x512 pixels (1x and 2x), 1024x1024 pixels (1x only) which will be converted to a single .icns file for MacOs to display the most appropriate icon anywhere.

We can improve on that, but maybe in another work?

Reply:
> While it's not necessary, we can create an icon with rounded borders like the others in the dock image. I used the softened-icon-dark.png logo in the client/webserver/site/img folder.

I assume that's what @buck54321 meant by "standardized".  Probably should although this PR is fine to merge now IMO.

Reply:
Found time to design one on Figma yesterday: https://www.figma.com/file/ao9XWuHAKDUJWfq3w2kjgD/dexc-macOS-icon?type=design&node-id=0-1&t=204FpylWMx9s29kT-0
@buck54321, does it look good?

I don't know where design superpower came from cuz I'm really not good with design tools 😄 .